### ISSUE TITLE
For creating Custom Views Using ContentManager 

### ISSUE THREAD

**Issue by chaklasiyanikunj** [rokudev/SceneGraphDeveloperExtensions#37]
I read the Rokudev sample UX component inside a list and grid. Here used an XML file for passing a content. Let's take one example is PosterGrid. Is it possible to convert PosterGrid to SGDEX View?

I read [this ](https://github.com/rokudev/SceneGraphDeveloperExtensions/blob/master/documentation/4-SGDEX_Channel_Tutorial.md) documentation for Opening non-SGDEX view. But Still, I'm Confused. I'm not sure Is it possible or not. Is it possible any Roku example convert with SGDEX View?

**Comment by RokuChris** [comment]
Yes, you can use PosterGrid in a custom view. You can use any Roku SceneGraph components you like to build a custom view and then use that custom view with SGDEX. There is an example of that here, https://github.com/rokudev/SceneGraphDeveloperExtensions/tree/master/samples/8_Custom%2BScreen

That said, using a custom view with SGDEX is not the same thing as using the ContentManager to _populate_ a custom view. The current version of the ContentManager only supports custom views that contain a grid type component. We recommend RowList or ZoomRowList. We did not test it with PosterGrid, but you are free to try it. Documentation for that is here, https://github.com/rokudev/SceneGraphDeveloperExtensions/blob/master/documentation/6-ContentManager_with_Custom_Views_Guide.md

**Comment by chaklasiyanikunj** [comment]
Thank you for your reply. I will try. I test the [custom view](https://github.com/rokudev/SceneGraphDeveloperExtensions/blob/master/documentation/6-ContentManager_with_Custom_Views_Guide.md). and tried to print the onKeyEvent() in mainscene.brs file using like below.

```
function onKeyEvent(key as String, press as Boolean)
     ? "MainScene :: Key Event is about to execute - key = "key " press = " press
end function
```
But, Here I found false press value. I do not understand Why I am getting "press = false" value in MainScene.brs. But, I noticed another view Extends with the SGDEXComponent and custom grid view Extends With the group. So, I think not to handle the key with MainScene.brs. Is there any way to handle all key in MainScene.brs.

**Comment by chaklasiyanikunj** [comment]
Any update regarding this?

**Comment by RokuChris** [comment]
This is what one of our engineers had to say

> OnKeyEvent works in custom views but it will only get key events that were not handled by the current component in focus, i.e. a child grid node in case of custom grid view - this is by design in RSG.
>
> RSG grid nodes like RowList, ZoomRowlist etc. handle most of the key presses (d-pad keys, ff/rw) behind the scenes and they don't get exposed to OnKeyEvent outside this grid node, unless developer extends it (e.g. myRowList) and implements OnKeyEvent function in this extended node scope

**Comment by chaklasiyanikunj** [comment]
For Other View and component, When I press the remote button key, I found press value "true". whenever I release the remote button key, I found press value "false".
Here, I tested with a CustomZoomRowList component key handler in both file CustomZoomRowList.brs and MainScene.brs. But, In both files. I found a press value "false". I also take the same example in the sample folder UX component/list and grid. Here I tested key handler with RowList and ZoomRowList It's also working fine. 
As per your suggestion, I created a Custom View With PosterGrid View. using this custom component, I not found any issue with the key handler. if Focus is set in CustomZoomRowList. then why found press = false in CustomZoomRowList.brs file. I'm still confused with the key handler.
